bench/bench.ml

```diff
@@ -0,0 +1,484 @@
+module Stats = Index.Stats
+
+let src =
+  let open Metrics in
+  let open Stats in
+  let tags = Tags.[] in
+  let data t =
+    Data.v
+      [
+        int "bytes_read" t.bytes_read;
+        int "bytes_written" t.bytes_written;
+        int "merge" t.nb_merge;
+        int "replace" t.nb_replace;
+      ]
+  in
+  Src.v "bench" ~tags ~data
+
+let key_size = 32
+
+let value_size = 13
+
+let entry_size = key_size + value_size
+
+let ( // ) = Filename.concat
+
+let random_char () = char_of_int (33 + Random.int 94)
+
+let random_string string_size =
+  String.init string_size (fun _i -> random_char ())
+
+module Context = struct
+  module Key = struct
+    type t = string
+
+    let v () = random_string key_size
+
+    let hash = Hashtbl.hash
+
+    let hash_size = 30
+
+    let encode s = s
+
+    let decode s off = String.sub s off key_size
+
+    let encoded_size = key_size
+
+    let equal = String.equal
+
+    let pp s = Fmt.fmt "%s" s
+  end
+
+  module Value = struct
+    type t = string
+
+    let v () = random_string value_size
+
+    let encode s = s
+
+    let decode s off = String.sub s off value_size
+
+    let encoded_size = value_size
+
+    let pp s = Fmt.fmt "%s" s
+  end
+end
+
+let with_stats f =
+  Stats.reset_stats ();
+  let t0 = Sys.time () in
+  let x = f () in
+  let t1 = Sys.time () -. t0 in
+  let stats = Stats.get () in
+  (x, t1, stats)
+
+module Benchmark = struct
+  type result = {
+    time : float;
+    ops_per_sec : float;
+    mbs_per_sec : float;
+    read_amplification_calls : float;
+    read_amplification_size : float;
+    write_amplification_calls : float;
+    write_amplification_size : float;
+  }
+  [@@deriving yojson]
+
+  let run ~nb_entries f =
+    let _, time, stats = with_stats (fun () -> f ()) in
```

Comment:
If this is the only place where you call `with_stats`, you don't need to return a triple then.

Reply:
Solved! :heavy_check_mark: